src/components/Account/AccountDetails.vue

```diff
@@ -0,0 +1,177 @@
+        type="date"
+        :rules="dateOfBirthRules"
+        @updateValidation="(value) => (user.date_of_birth.isValid = value)"
+      />
+      <CustomInput
+        ref="passwordInput"
+        class="account-details__input"
+        v-model="password"
+        label="Enter your password to enable save"
+        placeholder="Enter your password"
+        type="password"
+      />
+      <CustomButton
+        outlined
+        size="56"
+        class="account-details__button"
+        width="100%"
+        :disabled="!enableSave"
+        type="submit"
+        >Save changes</CustomButton
+      >
+    </form>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import CustomInput from "@/components/common/CustomInput.vue";
+import CustomButton from "@/components/common/CustomButton.vue";
+import { computed, onBeforeMount, Ref, ref } from "vue";
+import {
+  isEmailValid,
+  minLength,
+  isOldEnough,
+  ValidationRule,
+} from "@/helpers/validationRules";
+import { retrieveCurrentUser, updateCurrentUser } from "@/api/useAuthApi";
+import { notify } from "@kyvg/vue3-notification";
+
+interface UserDetail {
+  inputValue: string;
+  isValid: boolean;
+}
+interface User {
+  email: UserDetail;
+  first_name: UserDetail;
+  last_name: UserDetail;
+  date_of_birth: UserDetail;
+}
+const user: Ref<User | null> = ref(null);
+
+const fields: Array<string> = [
+  "email",
+  "first_name",
+  "last_name",
+  "date_of_birth",
+];
+async function getCurrentUserDetails() {
+  const retrievedUser = await retrieveCurrentUser();
+  user.value = fields.reduce(
+    (userInputs: User, property) => {
+      userInputs[property] = {
+        inputValue: retrievedUser[property],
+        isValid: true,
+      };
+      return userInputs;
+    },
+    {
+      email: { inputValue: "", isValid: false },
+      first_name: { inputValue: "", isValid: false },
+      last_name: { inputValue: "", isValid: false },
+      date_of_birth: { inputValue: "", isValid: false },
+    }
+  );
+}
+
+onBeforeMount(async () => {
+  await getCurrentUserDetails();
+});
+
+const password: Ref<string> = ref("");
+
+const emailRules: Array<ValidationRule> = [isEmailValid(false)];
+const nameRules: Array<ValidationRule> = [minLength(3, false)];
+const dateOfBirthRules: Array<ValidationRule> = [isOldEnough(18)];
+
+type CustomInputRef = InstanceType<typeof CustomInput> | null;
+const emailInput = ref<CustomInputRef>(null);
+const firstNameInput = ref<CustomInputRef>(null);
+const lastNameInput = ref<CustomInputRef>(null);
+const dateOfBirthInput = ref<CustomInputRef>(null);
+
+const enableSave = computed<boolean>(() => password.value.length >= 8);
+
+async function saveChanges() {
+  const updateForm = {};
+  fields.forEach(
+    (property) => (updateForm[property] = user.value[property].inputValue)
+  );
+  const status: number = await updateCurrentUser({
+    ...updateForm,
+    current_password: password.value,
+  });
+  if (status > 200 && status < 300) {
```

Comment:
1. `200` is also a success code
2. and what if it's not ok? 😁

Reply:
I think the request if successfull returns 201, probably the reason why sth between 200 and 300
You're right, 200 should be included
And again you're right I didn't handle the error message. I will make a general notification for all the errors in axios config for now.